docs/specification-ligo.md

```diff
@@ -0,0 +1,897 @@
+  ; decision_lambda : proposal * storage -> operation list * storage
+  // ^ The decision lambda is executed based on a successful proposal.
+  // It has access to the proposal, can modify `contractExtra` and perform arbitrary
+  // operations.
+
+  ; max_proposals : nat
+  // ^ Determine the maximum number of ongoing proposals that are allowed in the contract.
+  ; max_votes : nat
+  // ^ Determine the maximum number of votes associated with a proposal including positive votes
+  ; max_quorum_threshold : nat
+  // ^ Determine the maximum value of quorum threshold that is allowed to be set.
+  ; min_quorum_threshold : nat
+  // ^ Determine the minimum value of quorum threshold that is allowed to be set.
+  ; max_voting_period : nat
+  // ^ Determine the maximum value of voting period that is allowed to be set.
+  ; min_voting_period : nat
+  // ^ Determine the minimum value of voting period that is allowed to be set.
+
+  ; custom_entrypoints : custom_entrypoints
+  // ^ Packed arbitrary lambdas associated to a name for custom execution.
+  }
+```
+
+Note:
+- the `token_metadata` type matches the one defined in FA2.
+- the `proposal` type is defined below.
+- `storage` is the storage type of the contract without the configuration.
+- `full_storage` is instead the full storage of the contract, including its configuration,
+which is to say: `type full_storage = storage * config`.
+
+
+```cameLIGO
+type proposal_key = bytes
+
+type proposal =
+  { upvotes : nat
+  ; downvotes : nat
+  ; start_date : timestamp
+  ; metadata : proposal_metadata
+  ; proposer : address
+  ; proposer_frozen_token : nat
+  ; voters : (address * nat) list
+  // ^ List of voter addresses associated with the vote amount
+  // Needed for `flush` entrypoint.
+  }
+```
+
+## Runtime configuration
+
+Some configuration values are specified in runtime and can be changed during the contract lifetime.
+They must be provided on origination to construct the contract's initial storage.
+These values are:
+1. `admin : address` is the address that can perform administrative actions.
+2. `voting_period : nat` specifies how long the voting period lasts.
+3. `quorum_threshold : nat` specifies how many total votes are required for a successful proposal.
+
+# Contract logic
+
+This chapter provides a high-level overview of the contract's logic.
+
+- The contract maintains a ledger of address and its balance (frozen and unfrozen tokens)
+- The contract manages a special role called "Administrator".
+- The contract stores a list of proposals that can be in one of the states: "ongoing", "rejected", or "accepted".
+- Migration form a two-step process:
+  + After migration start the contract `migration_status` state is set to `MigratingTo targetAddress` state.
+  + After successful migration confirmation `migration_status` is set to `MigratedTo newContractAddress` state. All the subsequent operations, except `transfer_contract_tokens`, will fail with the `MIGRATED` tag.
```

Comment:
```suggestion
- Migration forms a two-step process:
  + After the migration starts, the contract `migration_status` state is set to `MigratingTo targetAddress` state.
  + After successful migration confirmation, `migration_status` is set to `MigratedTo newContractAddress` state. All the subsequent operations, except `transfer_contract_tokens`, will fail with the `MIGRATED` tag.
```

Reply:
Fixed here and in `specification.md` as well